ui/src/components/UserMenu/index.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component, Fragment } from 'react';
+import { Link } from 'react-router-dom';
+import { withApollo } from 'react-apollo';
+import { withStyles } from '@material-ui/core/styles';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import AccountIcon from 'mdi-react/AccountIcon';
+import HandPeaceIcon from 'mdi-react/HandPeaceIcon';
+import { withAuth } from '../../utils/Auth';
+import UserMenuList from './UserMenuList';
+import UserMenuButton from './UserMenuButton';
+
+@withStyles(theme => ({
+  avatar: {
+    backgroundColor: theme.palette.secondary.main,
+  },
+  userMenu: {
+    [theme.breakpoints.up('sm')]: {
+      padding: `${theme.spacing.unit / 2}px ${theme.spacing.double}px`,
+    },
+  },
+  text: {
+    color: theme.palette.text.primary,
+    fontFamily: 'Roboto500',
+  },
+  icon: {
+    fill: theme.palette.text.primary,
+  },
+  leftIcon: {
+    marginRight: theme.spacing.unit,
+  },
+  username: {
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis',
+  },
+}))
+@withAuth
+@withApollo
+export default class UserMenu extends Component {
+  state = {
+    anchorEl: null,
+    signInDialogOpen: false,
+  };
+
```

Comment:
Define a `static propTypes` and `static defaultProps` methods. See the other component files for inspiration.